www/app/Http/Controllers/MemoryController.php

```
@@ -0,0 +1,184 @@
+                return redirect()->back()->with('error', $result['message']);
+            }
+        } catch (\Exception $e) {
+            Log::error('Failed to create snapshot', ['error' => $e->getMessage()]);
+            return redirect()->back()->with('error', 'Failed to create snapshot');
+        }
+    }
+
+    /**
+     * Restore from a snapshot.
+     */
+    public function restoreSnapshot(Request $request, string $filename)
```

PHPMD (2.15.0)

Avoid unused parameters such as '$request'. (undefined)

(UnusedFormalParameter)